addons/hr_holidays/models/hr_leave.py

```diff
@@ -958,6 +919,24 @@ def _check_double_validation_rules(self, employees, state):
 
     @api.model_create_multi
     def create(self, vals_list):
+        employee_ids = [values['employee_id'] for values in vals_list if values.get('employee_id')]
+        employees = self.env['hr.employee'].browse(employee_ids)
+        if self._context.get('leave_compute_date_from_to') and employees:
+            for values in vals_list:
+                employee_id = values.get('employee_id')
+                if values.get('date_from') and values.get('date_to') and employee_id:
```

Comment:
As you already have the employee, check that first so it can skip getting the dates if there are no employee set

```suggestion
                if employee_id and values.get('date_from') and values.get('date_to'):
```